### ISSUE TITLE
[no-constant-condition] fails when both sides are explicitly coerced

### ISSUE THREAD

**Issue by alitaheri** [palantir/tslint#1662]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `nightly 2.1.0-dev.20161029`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const a = '1';
const b = '2';

// This is valid code, no constant condition here.
if (+a > +b) {
 console.log(a);
}
```

with `tslint.json` configuration:

```json
{
  "no-constant-condition": true,
}
```

#### Actual behavior

No warnings.

#### Expected behavior

`unexpected constant condition`

**Comment by alitaheri** [comment]
Sorry, this isn't related to ts-lint 😅
